opt_einsum/contract.py

```diff
@@ -414,13 +436,21 @@ def _core_contract(operands, contraction_list, **einsum_kwargs):
                 right_pos.append(input_right.find(s))
 
             # Contract!
-            new_view = np.tensordot(*tmp_operands, axes=(tuple(left_pos), tuple(right_pos)))
+            new_view = _tensordot(*tmp_operands, axes=(tuple(left_pos), tuple(right_pos)), backend=backend)
 
             # Build a new view if needed
             if (tensor_result != results_index) or handle_out:
+
+                transpose = tuple(map(tensor_result.index, results_index))
+
+                try:
+                    new_view = new_view.transpose(transpose)
+                except AttributeError:
+                    # some libraries don't implement method version
+                    new_view = _transpose(new_view, axes=transpose, backend=backend)
```

Comment:
Can we write this `try/expect` in `_transpose` as well to make it a bit more straight forward?

Reply:
Yep, that's definitely cleaner.